packages/docs/src/content/docs/reference/faq.md

```diff
@@ -0,0 +1,326 @@
+That's why Knip shares a single program for workspaces if their configuration
+allows it. This optimization is enabled by default, while it also allows the
+module resolver (one per program) to do some more caching.
+
+This can be disabled using the `--isolate-workspaces` flag, but is rarely
+necessary in practice.
+
+Also see [workspace sharing][3] (and [Slim down to speed up][4] for full
+history).
+
+## Why doesn't Knip just use `ts.findReferences`?
+
+TypeScript has a very good "Find references" feature, that you might be using in
+your IDE as well. Yet at scale this becomes too slow. That's why Knip builds up
+its own module graph to look up export usage a lot faster.
+
+Knip does use `ts.findReferences` to find references to class members (i.e. when
+`classMembers` is enabled).
+
+## Why can't I use path aliases to reference other workspaces?
+
+Some projects use `compilerOptions.paths` to alias paths to other workspaces in
+the same monorepo. This works for TypeScript and bundlers. However, it does not
+work well with Knip, since Knip doesn't understand those paths might represent
+workspaces. Knip is thus unable to match dependencies (including internal
+workspaces) in `package.json` against import usage correctly.
+
+Instead, it's recommended to list such workspaces/dependencies in
+`package.json`, and import them as such. TypeScript and bundlers have no issues
+with this standard approach either.
+
+## What's up with that configurable `tsconfig.json` location?
+
+There's a difference between `--tsConfig [file]` as a CLI argument and the
+`typescript.config` option in Knip configuration.
+
+The `--tsConfig [file]` option is used to provide an alternative location for
+the default root `tsconfig.json` file. Relevant `compilerOptions` include
+`paths` and `moduleResolution`. It's currently only possible to set this
+location ar the root level (i.e. not in other monorepo workspaces).
+
+On the other hand, the `typescript.config` option is a TypeScript plugin option,
+and can be set per-workspace. The plugin extracts referenced external
+dependencies such as those in `extends`, `compilerOptions.types` and JSX
+settings:
+
+```json title="tsconfig.json"
+{
+  "extends": "@tsconfig/node20/tsconfig.json",
+  "compilerOptions": {
+    "jsxImportSource": "hastscript/svg"
+  }
+}
+```
+
+From this example, Knip can determine whether the `@tsconfig/node20` and
+`hastscript` dependencies are properly listed in `package.json`.
+
+Note that the TypeScript plugin doesn't add support for TypeScript to Knip. Like
+other plugins, it extracts dependencies from configuration files. With the
+`typescript.config` option an alternative location for `tsconfig.json` can be
+set per workspace.
+
+## How does Knip handle non-standard import syntax?
+
+Knip tries to be resilient against import syntax like what's used by e.g.
+Webpack loaders or Vite asset imports. Knip strips off the prefixes and suffixes
```

Comment:
According to its documentation, `webpack` is not capitalized (https://webpack.js.org/guides/getting-started/), as shown for instance by this sentence in their docs.

> Let's use webpack to manage these scripts instead.

Reply:
Thanks, indeed